### ISSUE TITLE
memory leak: writev to closed pipe with SIGPIPE blocked

### ISSUE THREAD

**Issue by mikbras** [deislabs/mystikos#420]
```
2021-06-02T00:37:32.0168978Z === blocks in use: 144 bytes in 2 blocks
2021-06-02T00:37:32.0169762Z 16 bytes
2021-06-02T00:37:32.0170600Z 0x7ff2843eb89f: myst_debug_malloc()
2021-06-02T00:37:32.0175608Z 0x7ff2843eb9e2: myst_debug_calloc()
2021-06-02T00:37:32.0176118Z 0x7ff284408ca4: calloc()
2021-06-02T00:37:32.0176523Z 0x7ff28440bff9: myst_signal_deliver()
2021-06-02T00:37:32.0177064Z 0x7ff2843d7de2: myst_syscall_kill()
2021-06-02T00:37:32.0177464Z 0x7ff2843fd73f: _pd_write()
2021-06-02T00:37:32.0178300Z 0x7ff284427844: myst_fdops_writev()
2021-06-02T00:37:32.0178798Z 128 bytes
2021-06-02T00:37:32.0179232Z 0x7ff2843eb89f: myst_debug_malloc()
2021-06-02T00:37:32.0179639Z 0x7ff2843eb9e2: myst_debug_calloc()
2021-06-02T00:37:32.0180031Z 0x7ff284408ca4: calloc()
2021-06-02T00:37:32.0180422Z 0x7ff2843d7d9c: myst_syscall_kill()
2021-06-02T00:37:32.0180807Z 0x7ff2843fd73f: _pd_write()
2021-06-02T00:37:32.0181201Z 0x7ff284427844: myst_fdops_writev()
2021-06-02T00:37:32.0181520Z 
2021-06-02T00:37:32.0181867Z *** memory leaks detected
```

**Comment by vtikoo** [comment]
Stripped down program which has this leak -
```
#include <unistd.h>
#include <signal.h>
#include <sys/uio.h>

#define	CHUNK		64
static char buf[CHUNK * 4];
struct iovec iovec_simple[] = {
	{ buf, CHUNK },
};
# define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))
int main()
{
    sigset_t block_mask;

	sigemptyset(&block_mask);
	sigaddset(&block_mask, SIGPIPE);
	sigprocmask(SIG_BLOCK, &block_mask, NULL);

    int pipefd[2];
    pipe(pipefd);
    close(pipefd[0]);
    writev(pipefd[1], iovec_simple, ARRAY_SIZE(iovec_simple));
}
```

**Comment by vtikoo** [comment]
Here the thread is blocking SIGPIPE and then writing to a closed pipe. The signal is never unblocked and the main thread exits.
This causes the siginfo_list_item object for this signal to stay around in `thread.signal.siginfos`.
cc: @jxyang

**Comment by vtikoo** [comment]
#506 tries to fix by freeing thread.signal.siginfos during thread shutdown in myst_signal_free.

**Comment by vtikoo** [comment]
Closed by #506
